lib/internal/test_runner/utils.js

```diff
@@ -168,7 +169,10 @@ async function setupTestReporters(rootTest) {
   const reportersMap = await getReportersMap(reporters, destinations, rootTest);
   for (let i = 0; i < reportersMap.length; i++) {
     const { reporter, destination } = reportersMap[i];
-    compose(rootTest.reporter, reporter).pipe(destination);
+    compose(rootTest.reporter, reporter).on('error', (err) => {
+      process.exitCode = 1;
```

Comment:
```suggestion
      process.exitCode ||= 1;
```
in case it's already been set to something else nonzero that's meaningful?